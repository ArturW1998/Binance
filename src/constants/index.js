export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
