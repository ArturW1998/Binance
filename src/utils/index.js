import { LISTS } from '../constants';

export const getTableDataListWithValues = (
  symbol,
  latestPrice,
  openPrice,
  highPrice,
  lowPrice,
  quoteVolume,
) => {
  const { TABLE_DATA_LIST } = LISTS;

  return TABLE_DATA_LIST.map(item => {
    switch (item.title) {
      case 'Pair':
        return { ...item, value: symbol, bold: true };
      case 'Latest Price':
        return { ...item, value: latestPrice };
      case 'Open':
        return { ...item, value: openPrice };
      case 'High':
        return { ...item, value: highPrice };
      case 'Low':
        return { ...item, value: lowPrice };
      case 'Volume':
        return { ...item, value: quoteVolume };
      default:
        return { ...item };
    }
  });
};

export const getTickerBySymbol = (data = []) => {
  const ticker = {};

  data.forEach(
    ({
      s: symbol,
      c: latestPrice,
      p: priceChange,
      P: priceChangePercent,
      h: highPrice,
      l: lowPrice,
      q: quoteVolume,
      o: openPrice,
    }) => {
      ticker[symbol] = {
        symbol,
        latestPrice,
        priceChange,
        priceChangePercent,
        highPrice,
        lowPrice,
        quoteVolume,
        openPrice,
      };
    },
  );

  return ticker;
};
