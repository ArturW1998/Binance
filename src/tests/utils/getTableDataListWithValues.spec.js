import { LISTS } from '../../constants';
import { getTableDataListWithValues } from '../../utils';

const { TABLE_DATA_LIST } = LISTS;

const symbol = 'BNBBTC';
const latestPrice = '0.00184990';
const openPrice = '0.00187580';
const highPrice = '0.00182500';
const lowPrice = '2332.46079005';
const quoteVolume = '0.00184780';

describe('getTableDataListWithValues', () => {
  it('getTableDataListWithValues with TABLE_DATA_LIST', () => {
    const expectedData = [
      { title: 'Pair', value: symbol, bold: true },
      { title: 'Latest Price', value: latestPrice },
      { title: 'Open', value: openPrice },
      { title: 'High', value: highPrice },
      { title: 'Low', value: lowPrice },
      { title: 'Volume', value: quoteVolume },
    ];

    expect(
      getTableDataListWithValues(
        TABLE_DATA_LIST,
        symbol,
        latestPrice,
        openPrice,
        highPrice,
        lowPrice,
        quoteVolume,
      ),
    ).toEqual(expectedData);
  });

  it('getTableDataListWithValues with TABLE_DATA_LIST toMatchSnapshot', () => {
    expect(
      getTableDataListWithValues(
        TABLE_DATA_LIST,
        symbol,
        latestPrice,
        openPrice,
        highPrice,
        lowPrice,
        quoteVolume,
      ),
    ).toMatchSnapshot();
  });
});

describe('getTableDataListWithValues initial state', () => {
  it('should return the initial state', () => {
    expect(getTableDataListWithValues()).toEqual(TABLE_DATA_LIST);
  });

  it('returns properly', () => {
    expect(getTableDataListWithValues()).toMatchSnapshot();
  });
});

describe('getTableDataListWithValues fake state', () => {
  const list = [
    { title: 'Baase asset' },
    { title: 'Quote asset' },
    { title: 'Parent market' },
    { title: 'Category of the parent market' },
  ];

  it('getTableDataListWithValues with fake data', () => {
    expect(
      getTableDataListWithValues(
        list,
        symbol,
        latestPrice,
        openPrice,
        highPrice,
        lowPrice,
        quoteVolume,
      ),
    ).toEqual(list);
  });

  it('getTableDataListWithValues with fake data toMatchSnapshot', () => {
    expect(
      getTableDataListWithValues(
        list,
        symbol,
        latestPrice,
        openPrice,
        highPrice,
        lowPrice,
        quoteVolume,
      ),
    ).toMatchSnapshot();
  });
});
