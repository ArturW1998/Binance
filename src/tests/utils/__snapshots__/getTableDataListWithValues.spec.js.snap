// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTableDataListWithValues fake state getTableDataListWithValues with fake data toMatchSnapshot 1`] = `
Array [
  Object {
    "title": "Baase asset",
  },
  Object {
    "title": "Quote asset",
  },
  Object {
    "title": "Parent market",
  },
  Object {
    "title": "Category of the parent market",
  },
]
`;

exports[`getTableDataListWithValues getTableDataListWithValues with TABLE_DATA_LIST toMatchSnapshot 1`] = `
Array [
  Object {
    "bold": true,
    "title": "Pair",
    "value": "BNBBTC",
  },
  Object {
    "title": "Latest Price",
    "value": "0.00184990",
  },
  Object {
    "title": "Open",
    "value": "0.00187580",
  },
  Object {
    "title": "High",
    "value": "0.00182500",
  },
  Object {
    "title": "Low",
    "value": "2332.46079005",
  },
  Object {
    "title": "Volume",
    "value": "0.00184780",
  },
]
`;

exports[`getTableDataListWithValues initial state returns properly 1`] = `
Array [
  Object {
    "title": "Pair",
  },
  Object {
    "title": "Latest Price",
  },
  Object {
    "title": "Open",
  },
  Object {
    "title": "High",
  },
  Object {
    "title": "Low",
  },
  Object {
    "title": "Volume",
  },
]
`;
